code/game/atoms_movable.dm

```diff
@@ -148,17 +148,31 @@
 	return 1
 
 /mob/living/forceMove(atom/destination)
-	stop_pulling()
 	if(buckled)
-		buckled.unbuckle_mob(src,force=1)
-	// in lieu of "unbuckle_all_mobs"
+		addtimer(src, "checkBuckled", 1, TRUE)
 	if(buckled_mob)
-		unbuckle_mob(buckled_mob,force=1)
+		addtimer(src, "checkBuckledMob", 1, TRUE)
+	if(pulling)
+		addtimer(src, "checkPull", 1, TRUE)
 	. = ..()
 	if(client)
 		reset_perspective(destination)
 	update_canmove() //if the mob was asleep inside a container and then got forceMoved out we need to make them fall.
 
+/mob/living/proc/checkBuckled()
+	if(!(buckled in loc))
+		buckled.unbuckle_mob(src, force=1)

```

Comment:
No sanity check for `buckled` becoming null before the timer fires.

Also, this proc probably belongs with the rest of the buckling code.